Add sample module for Apigee Provisioning without VPC Peering

baskaran-md opened the pull request:
[Apigee announced the GA launch of its non-VPC provisioning option](https://cloud.google.com/apigee/docs/release-notes#August_26_2024)

- Add sample modules for provisioning Apigee without VPC Peering. 
   - Made network & ip_ranges as optional variables in `apigee-x-core` module.

- [x] I tested the provisioning locally and is successful. 
- [x] I have followed the relevant style guide for my changes.

g-greatdevaks commented:
Thanks for the PR, Baskaran.
Let me review and merge this.

danistrebel commented on modules/apigee-x-core/variables.tf:
```diff
@@ -36,9 +36,16 @@ variable "ax_region" {
 }
 
+variable "disable_vpc_peering" {
+  description = "Flag that specifies whether the VPC Peering through Private Google Access should be disabled between the consumer network and Apigee."
```
Shouldn't this say through "Service Networking"? PGA is something different.

baskaran-md replied:
My bad. It should be "Private Service Access". Mentioning SN as well.

https://cloud.google.com/vpc/docs/configure-private-services-access

danistrebel commented on modules/apigee-x-core/variables.tf:
```diff
@@ -36,9 +36,16 @@ variable "ax_region" {
+}
+
 variable "network" {
   description = "Network (self-link) to peer with the Apigee tennant project."
```
Can you also mention that his isn't needed for the non-peering mode

baskaran-md replied:
Done

danistrebel commented on samples/x-non-vpc-peering/README.md:
```diff
@@ -0,0 +1,80 @@
+# Basic Apigee X Setup without VPC Peering
+
+A minimal Apigee X setup without using VPC Peering.
```
Maybe worth adding a link to the docs or at least the release notes that explain this.

baskaran-md replied:
Done.

danistrebel commented on samples/x-non-vpc-peering/README.md:
```diff
@@ -0,0 +1,80 @@
+# Basic Apigee X Setup without VPC Peering
```
As the samples are meant to be somewhat self sufficient shouldn't this module also contain a northbound PSC component? Otherwise it could also just be a doc of setting networking to false in the x-core module.

baskaran-md replied:
Done. Adding NB XLB + PSC.

danistrebel reviewed (changes requested):
Thanks for the PR! 

Definitely good to add the options to the core module. For the sample I would suggest maybe adding PSC + ILB setup or at least mentioning it in the README

The  following would be needed in addition to the comments:

- [ ] simple test for the sample see for example the test for x-basic sample here https://github.com/apigee/terraform-modules/blob/main/tests/samples/test_x_basic.py
- [ ] re-create the docs for all modules with the script in [./tools/format-repo.sh](./tools/format-repo.sh)

baskaran-md commented:
@danistrebel / @g-greatdevaks - Please review the PR again.

- [X] Added PSC + XLB setup.
- [X] Added test @ tests/samples/test_x_non_vpc_peering.py
- [X] Recreated the docs using re-create the docs for all modules with the script in [./tools/format-repo.sh](https://github.com/apigee/terraform-modules/pull/tools/format-repo.sh). FYI - It formatted few other files that I didn't modify as well.

danistrebel commented:
Thanks @baskaran-md ,

The last thing that the pipeline complains about in https://github.com/apigee/terraform-modules/actions/runs/11522276251/job/32260268126?pr=159 are the following README files.

```
The Documentation in the following README files is out of date:
/github/workspace/modules/apigee-x-core/README.md
/github/workspace/samples/x-non-vpc-peering/README.md
```

Do you mind re-running the TF docs tool for these two modules?

otherwise all LGTM!

baskaran-md commented:
- [x] Ran both `sh tools/update-docs.sh` & `sh tools/format-repo.sh`. Pushed the updated docs. 
- [x] Updated main README.md to include this new sample to the list.

Thank you for reviewing.

baskaran-md reviewed (commented):
- [x] Ran both `sh tools/update-docs.sh` & `sh tools/format-repo.sh`. Pushed the updated docs. 
- [x] Updated main README.md to include this new sample to the list.

Thank you for reviewing.
